### PR TITLE
Pure js edit podcast

### PR DESCRIPTION
**Pull request by nirjacob**

### CODE REVIEW

File: client/public/editPodcastPage/editPodcastPage.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    <script src="./editPodcastPage.js"></script>
+</head>
+<body>
+<div class="container">
+    <div class="container-middle">
+        <div class="form-title">Edit Podcast Information</div>
+        <label class="form-labels">Title</label>
+        <input class="form-button" type="text" id="title" value="Title Placeholder">
+        <label class="form-labels">Description</label>
+        <input class="form-button" type="text" id="description" value="Description Placeholder">
```
Comment (racheli-lavan) [inline client/public/editPodcastPage/editPodcastPage.html:16]: for a field like description I would use textarea and not input element

File: client/public/editPodcastPage/editPodcastPage.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        <label class="form-labels">Html Description</label>
+        <input class="form-button" type="text" id="htmlDescription" value="Html Description Placeholder">
```
Comment (racheli-lavan) [inline client/public/editPodcastPage/editPodcastPage.html:18]: same here

File: client/public/editPodcastPage/editPodcastPage.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        <label class="form-labels">Web Url</label>
+        <input class="form-button" type="text" id="webUrl" value="Web Url Placeholder">
+        <label class="form-labels">Image Url</label>
+        <input class="form-button" type="text" id="imageUrl" value="Image Url Placeholder">
+        <label class="form-labels">Language</label>
+        <input class="form-button" type="text" id="language" value="Language Placeholder">
+        <label class="form-labels">Number Of Episodes</label>
+        <input class="form-button" type="text" id="numberOfEpisodes" value="1">
```
Comment (racheli-lavan) [inline client/public/editPodcastPage/editPodcastPage.html:26]: the type here should be number

File: client/public/editPodcastPage/editPodcastPage.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <label class="form-labels">Avg Episode Length</label>
+        <input class="form-button" type="text" id="avgEpisodeLength" value="1">
```
Comment (racheli-lavan) [inline client/public/editPodcastPage/editPodcastPage.html:28]: also number

File: client/public/editPodcastPage/editPodcastPage.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+window.onload = (event) => {
+  const deleteButton = document.getElementById('delete-btn')
+  deleteButton.parentNode.removeChild(deleteButton)
+}
+const editPodcast = async () => {
```
Comment (racheli-lavan) [inline client/public/editPodcastPage/editPodcastPage.js:5]: The name here should be savePodcast I believe

File: client/public/editPodcastPage/editPodcastPage.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  const editedPodcast = {
```
Comment (racheli-lavan) [inline client/public/editPodcastPage/editPodcastPage.js:6]: What happens if the user doesn't fill part of the fields and leave them empty? then an empty value will be updated?

File: client/public/editPodcastPage/editPodcastPage.html (hunk 1 above)
Comment (nirjacob) [reply client/public/editPodcastPage/editPodcastPage.html:16]: Fixed

File: client/public/editPodcastPage/editPodcastPage.html (hunk 2 above)
Comment (nirjacob) [reply client/public/editPodcastPage/editPodcastPage.html:18]: Fixed

File: client/public/editPodcastPage/editPodcastPage.html (hunk 4 above)
Comment (nirjacob) [reply client/public/editPodcastPage/editPodcastPage.html:28]: Fixed

File: client/public/editPodcastPage/editPodcastPage.js (hunk 5 above)
Comment (nirjacob) [reply client/public/editPodcastPage/editPodcastPage.js:5]: Changed.

File: client/public/editPodcastPage/editPodcastPage.js (hunk 6 above)
Comment (nirjacob) [reply client/public/editPodcastPage/editPodcastPage.js:6]: In the real app we would have the podcast details already as the fields value.
There is an edge case that the admin deletes a field and send it empty (which could be on purpose), I can handle it by having a formatting function that deletes empty fields, the question is if its necessary design wise?

Comment (nirjacob) [reply client/public/editPodcastPage/editPodcastPage.js]: I ended up adding a on-liner that filters them.

**Review by racheli-lavan — APPROVED** [review]

**Pull request merged**